lib/finder_game/util/overlay_renderer.dart

```diff
@@ -0,0 +1,126 @@
+import 'package:flame/extensions.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_game_challenge/finder_game/const/finder_constraints.dart';
+import 'package:flutter_game_challenge/finder_game/util/overlay_mask_calculator.dart';
+import 'package:flutter_game_challenge/finder_game/util/overlay_render_mode.dart';
+
+class OverlayRenderer {
```

Comment:
Why could not it be a `Flame Component `? This class already has methods like `resize` and `render`, looks like a candidate to a component.

Reply:
Because most of Component class features will not be used, and this class is tightly coupled with overlay in finder, so I don't see a point of using Component here.